### ISSUE TITLE
jspm 0.17 beta: "jspm link" says package has no main entry point

### ISSUE THREAD

**Issue by jdanyow** [jspm/jspm-cli#1666]
I'm in the `app` directory and executing `jspm link ../twilio`, which results in:
![command](http://i.imgur.com/MmoIMmS.png)

But I do have a `main` configured in the `twilio` package:
![folder structure](http://i.imgur.com/yF26E5J.png)

I'm attempting to apply these instructions to a project:

>If the `name`, `version` and `registry` had not been provided at all, jspm would default to linking as local:dirname@master, which enables a workflow to install private local folders. For packages that will be published, the linked name should typically always match the name that will be published as this allows us to easily switch between the local and remote versions without having to change package names.
http://jspm.io/0.17-beta-guide/linking.html

Related issue- not sure if it's a bug but `jspm install --unlink` doesn't work for private packages.  Is there an alternate command that will undo the `jspm link ../twilio` command?  I get the following error:
![unlink error](http://i.imgur.com/bTzxsJg.png)

If I use `jspm uninstall overpass-twilio` it uninstalls successfully but the package.json sym link is left in `twilio\dist\amd` which causes the following error if I re-run `jspm link ../twilio`
![error unlinking and re-linking](http://i.imgur.com/tgTHhhW.png)

**Comment by jdanyow** [comment]
I think I've figured this out.  I bet this is windows specific.  On my (windows 10) machine, the sym link for the package.json is a **directory** style link.  If I delete it and execute the following:

```shell
mklink package.json "C:\repos\overpass\twilio\package.json"
```

I can then re-run the `jspm link ../twilio` command and everything works.

It's as-if the package.json link is created using the equivalent of:
```shell
mklink /D package.json "C:\repos\overpass\twilio\package.json"
```

Tracked this down to [here](https://github.com/jspm/jspm-cli/blob/0.17/lib/link.js#L139) in the jspm cli code- `'junction'` is passed for the `type` arg. 

Maybe changing it to `'file'` would fix things for windows users without harming everyone else?

>fs.symlink(target, path[, type], callback)#
Asynchronous symlink(2). No arguments other than a possible exception are given to the completion callback. The type argument can be set to 'dir', 'file', or 'junction' **(default is 'file') and is only available on Windows (ignored on other platforms).** Note that Windows junction points require the destination path to be absolute. When using 'junction', the target argument will automatically be normalized to absolute path.
https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback

**Comment by guybedford** [comment]
Thanks for taking the effort to track this down. That sounds like it is exactly the issue. I've added https://github.com/jspm/jspm-cli/commit/b71e00da9d439b68ef78fec9eedcac32489fdaec for the next release here.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.13.
